.tidyrc.json

```diff
@@ -0,0 +1,10 @@
+{
+  "importSort": "ide",
+  "importWrap": "source",
+  "indent": 2,
+  "operatorsFile": null,
+  "ribbon": 5,
```

Comment:
FWIW, this ribbon value doesn't make sense. Ribbon should be a value between 0 and 1, and it represents the ratio of printable area to remaining width (given indentation). So a width of 160, and a ribbon of 0.5 would give you a printable area of 80 columns given no indentation. At an indentation of say 40, you would have a printable area of 60 additional columns (100 total width).

Reply:
My personal suggestion is to not use a formatter-enforced width and ribbon for hand-written code. It primarily exists for accessibility and code-generation.

Here's tidy-codegen's config for code-generation: https://github.com/natefaubion/purescript-tidy-codegen/blob/main/src/Tidy/Codegen.purs#L173-L183

Reply:
I think, formatter-enforced width standardizes, it also makes code more readable, e.g., it doesn't encourage introducing deeply nested blocks/constructs.

Reply:
Actually glancing over the changes there are arbitrary line breaks introduced, even on lines that are shorter than those surrounding, that just make code harder to read.

Reply:
I don't think there are line breaks on lines that are shorter than the desired width, it's just if you have something like a function application with a bunch of arguments, it will break the arguments on multiple lines at the point of overflow, so it can sometimes look that way. But, yeah, I agree it doesn't look more readable, which is why I don't recommend it. All of the configurations in `contrib` use default formatter settings for width/ribbon.

Reply:
Yes, in some places automatic formatting made things look strange, I will have a look through the sources more carefully and add some manual edits to the commit.

@nwolverson can you point out some of those places with _arbitrary_ line breaks you see?